### ISSUE TITLE
control legend format?

### ISSUE THREAD

**Issue by ellisp** [mtennekes/treemap#31]
would you consider a pull request that let the user control the legend format? eg pass through arguments to `format` so it can have big.marks, etc.  I could have a go but won't put the effort in if you've already thought it through and decided there's enough customisation and long argument lists already.

**Comment by mtennekes** [comment]
Isn't this already implemented in the argument `legend.format`, or do you still miss certain functionalities?

**Comment by mtennekes** [comment]
I was having tmap in mind, sorry:) 
Indeed, it's not in treemap yet. Please feel free to push a request.

**Comment by ellisp** [comment]
I've lodged https://github.com/mtennekes/treemap/pull/32 to address this.  I see I have rights to merge it myself (I think this dates from years ago when you gave me those rights for additions re fonts), but I've refrained from doing this until I have @mtennekes views on the naming, and whether it's right to pass it only to `dens2col` and `val2col` legends (nearly all my treemaps are type "value" so I'm a bit hazy on the best treatment of the other types).

**Comment by mtennekes** [comment]
Many thanks @ellisp ! The implementation seems correct to me. The value, density and manual treemaps are indeed the ones that could use formatting. As for the other ones: comparison does not need it since it contain percentages, and the others do not have numeric values. 

I have updated the DESCRIPTION file. Is you email address correct? Do you want me to release it soon? Normally, I wait under there are at least a couple of changes.

**Comment by ellisp** [comment]
Great, thanks.  I changed the email address to my personal one; sometimes difficult to tell which hat I'm wearing, but this one was done at home in my time so probably best to put it like that (although we need it at work).  No rush to put it into production at CRAN, put out a release when it makes sense.

**Comment by Atan1988** [comment]
Is it possible to format the number as percentages? I generally use sprintf for stuff like that, and I am not aware how to do it using format().  

Thanks,
Allen

**Comment by ellisp** [comment]
@Atan1988 I don't think it will let you do that.  That would be another feature to add I think; either to replace `format` with `sprintf`` (which would not be backwards compatible with #32 so if it's going to happen should ideally be soon)  or additional arguments "suffix.legend" and "prefix.legend".  As a user I'd find the latter slightly easier to understand but it  comes with extra arguments and loss of flexibility.

**Comment by Atan1988** [comment]
@ellisp  Peter, thank you for the response and the great work in implementing the format function.  I know R fairly well, but I haven't learned how to use git yet.  Would you have time to implement the sprinf function? If not, would you be willing to help with the pull/commit side of the things, if I take a shot at modifying the code? Thanks a lot! Allen

**Comment by mtennekes** [comment]
It's really confusing, all those formatting functions: `format`, `sprintf`, `formatC`, `prettyNum` :-s

As @ellisp suggested, we can add `suffix`  and `prefix` to the input list of `format.legend` (while the other arguments are passed on to `format`). Additionally, we can add another list argument, called `FUN`, which determines to which formatting function is used (by default `format`).

**Comment by benjaminrobinson** [comment]
Was looking through the issue statement and the percentage issue is something that would be really helpful for me...just my two cents.  Thanks for creating the package!
